internal/services/kubernetesconfiguration/arc_kubernetes_cluster_extension_resource.go

```diff
@@ -0,0 +1,262 @@
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/kubernetesconfiguration/2022-11-01/extensions"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type ArcKubernetesClusterExtensionModel struct {
+	Name                           string            `tfschema:"name"`
+	ResourceGroupName              string            `tfschema:"resource_group_name"`
+	ClusterName                    string            `tfschema:"cluster_name"`
+	ConfigurationProtectedSettings map[string]string `tfschema:"configuration_protected_settings"`
+	ConfigurationSettings          map[string]string `tfschema:"configuration_settings"`
+	ExtensionType                  string            `tfschema:"extension_type"`
+	ReleaseNamespace               string            `tfschema:"release_namespace"`
+	ReleaseTrain                   string            `tfschema:"release_train"`
+	TargetNamespace                string            `tfschema:"target_namespace"`
+	Version                        string            `tfschema:"version"`
+	CurrentVersion                 string            `tfschema:"current_version"`
+}
+
+type ArcKubernetesClusterExtensionResource struct{}
+
+var _ sdk.ResourceWithUpdate = ArcKubernetesClusterExtensionResource{}
+
+func (r ArcKubernetesClusterExtensionResource) ResourceType() string {
+	return "azurerm_arc_kubernetes_cluster_extension"
+}
+
+func (r ArcKubernetesClusterExtensionResource) ModelObject() interface{} {
+	return &ArcKubernetesClusterExtensionModel{}
+}
+
+func (r ArcKubernetesClusterExtensionResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return extensions.ValidateExtensionID
+}
+
+func (r ArcKubernetesClusterExtensionResource) Arguments() map[string]*pluginsdk.Schema {
+	arguments := commonArguments()
+	arguments["identity"] = commonschema.SystemAssignedIdentityRequiredForceNew()
+	return arguments
+
+}
+
+func (r ArcKubernetesClusterExtensionResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"current_version": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (r ArcKubernetesClusterExtensionResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ArcKubernetesClusterExtensionModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.KubernetesConfiguration.ExtensionsClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			id := extensions.NewExtensionID(subscriptionId, model.ResourceGroupName, "Microsoft.Kubernetes", "connectedClusters", model.ClusterName, model.Name)
```

Comment:
These shouldn't be hardcoded strings but enums in the swagger - we have a PR open for that so could we add a comment above referencing it:
```suggestion
			// defined as strings because they're not enums in the swagger https://github.com/Azure/azure-rest-api-specs/pull/23545
			id := extensions.NewExtensionID(subscriptionId, model.ResourceGroupName, "Microsoft.Kubernetes", "connectedClusters", model.ClusterName, model.Name)
```